### ISSUE TITLE
Sporadic multiproc test failures due to Popen bug

### ISSUE THREAD

**Issue by int19h** [microsoft/ptvsd#1914]
This is a Python bug:
https://bugs.python.org/issue37380

TL;DR: when a `subprocess.Popen` instance is GC'd, the process handle gets closed, but it is not removed from "active processes" list if the process in question was still running at the time. At some later point, attempting to create a _new_ `Popen` instance tries to clean up the active process list first thing in its `__init__` (invoking `subprocess._cleanup`, which invokes `subprocess._internal_poll` on everything in `subprocess._active`), and fails with "OSError: [WinError 6] The handle is invalid". This is not recoverable - from that point on, all attempts to spawn new subprocesses in that process will fail.

This is fixed in Python 3.9, and backported to 3.8 and 3.7. However, 2.7, 3.5, and 3.6 are all still susceptible to this.

In our test runs, this affects tests where the debuggee uses `subprocess` - `test_subprocess`, obviously, but also all the Flask tests.

The likely culprit is our use of `Popen` in `ptvsd.enable_attach()`, where we spawn the adapter. The adapter remains alive for as long as debuggee is, by design, but the `Popen` instance is referenced by a local, and goes away immediately. If so, we can keep it alive for longer by referencing it from a global.
